### PR TITLE
Banish build-eclipse

### PR DESCRIPTION
**Pull request by nik9000**
We don't need it any more.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -343,13 +343,13 @@ allprojects {
   }
 
   plugins.withType(JavaBasePlugin) {
-    eclipse.classpath.defaultOutputDir = file('build-eclipse')
```
Comment (nik9000) [inline build.gradle:346]: Eclipse actually didn't like how `build-eclipse` was both the default and contained the folders from below. I shoved the "default" folder to the side.

File: build.gradle [hunk 2]
```diff
@@ -343,13 +343,13 @@ allprojects {
...
+    eclipse.classpath.defaultOutputDir = file('build/eclipse/default')
     eclipse.classpath.file.whenMerged { classpath ->
       // give each source folder a unique corresponding output folder
       int i = 0;
       classpath.entries.findAll { it instanceof SourceFolder }.each { folder ->
         i++;
-        folder.output = "build-eclipse/" + i
+        folder.output = "build/eclipse/" + i
```
Comment (nik9000) [inline build.gradle:352]: This seems to work ok!

File: libs/core/.classpath1 [hunk 3]
```diff
@@ -1,340 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nik9000) [inline libs/core/.classpath1:1]: I don't believe we need this one.

**Review by nik9000 — COMMENTED** [review]
We'd talked about wanting to remove `build-eclipse` in the past and I figured I could blast it. I have not removed it from `.gitignore` for now so we don't accidentally commit garbage until everyone's regenerated their eclipse config.

File: build.gradle (hunk 1 above)
Comment (mark-vieira) [reply build.gradle:346]: What _is_ the actual default? Is there a reason to override this at all?

**Comment by nik9000** [comment]
I'll check. Probably `bin` or some garbage like that. It might not matter
to override because we set the build for everywhere.

File: build.gradle (hunk 1 above)
Comment (nik9000) [reply build.gradle:346]: I'll check. Probably `bin` or some garbage like that. It might not matter
to override because we set the build for everywhere.

Comment (nik9000) [reply build.gradle:346]: Its `bin/default`.

**Comment by nik9000** [comment]
OK! I've discovered why we have `build-eclipse` by running with this enabled for a bit. Eclipse freaks out if another process nukes is build files. It logs thousands of errors off in its "log" panel and shows thousands of "this class's hierarchy is broken" and "I can't find this class" issues.

**Review by mark-vieira — APPROVED** [review]
:+1:

**Comment by nik9000** [comment]
I don't think I can merge this until I figure out where to put the files so Eclipse is happy when we clean. What would you think of `out` like intellij?

**Comment by mark-vieira** [comment]
I don't have strong feelings on it. Whatever doesn't clash with other Eclipse output seems fine to me. Using `out` would work since we already have it listed in `.gitignore`.

**Comment by nik9000** [comment]
@mark-vieira - I've moved Eclipse into `out/eclipse/${source_set_number}` so we're consistent with IntelliJ.

**Comment by mark-vieira** [comment]
:+1:

**Pull request merged**